### ISSUE TITLE
Aliasing

### ISSUE THREAD

**Issue by gamemachine** [huwb/crest-oceanrender#83]
So with certain wave patterns there is a lot of very noticeable aliasing as you can see from the pic below.  V2 post processing doesn't help much (the pick is with smaa enabled).  Livenda's temporal anti aliasing can, but with temporal it's basically hope you have no fast moving objects, so not really viable.

This must be coming from something specific about the mesh in the crest, at least that's where I was going to start taking a look.  Any other ideas on this?  Short of don't make patterns with sharp crests?

![image](https://user-images.githubusercontent.com/75981/46234662-3893fc80-c32c-11e8-9493-2b272778117f.png)

**Comment by gamemachine** [comment]
Forgot to mention you don't get this looking at waves going away from you with foam enabled.  It's still there but it blends into the foam.  Wonder if that would be a simple solution, make the foam start just before the crest.  Actually I had a reaction to the foam in that regard anyways aliasing aside that it should start just slightly before the crest.

**Comment by gamemachine** [comment]
On second thought foam might not help, foam is being added a bit before there is something else going on that makes it different when looking at it front vs back that I can't pin down.

**Comment by huwb** [comment]
Yes there is aliasing at the silhouette of the waves. I cant speak to the front vs back, i've not noticed something like that but i can think of at least a few directional things that could affect it - foam, sun direction, env map orientation, ..

Re the aliasing, I have looked a little bit at the aliasing in the past but didnt get too far, and its an unsolved research problem and it currently appears to be a hard one. I think TAA will struggle with fast changing shading on the water surface and with motion as you note.

I'll leave this open for a couple of days to catch any other comments and mull it over.
